clcache.py

```diff
@@ -864,7 +864,11 @@ def parseArgumentsAndInputFiles(cmdline):
                     else:
                         raise AssertionError("Unsupported argument type.")
 
-                    arguments[arg.name].append(value)
+                    # consistently use empty list for flags like /c or /doc
+                    if not arg.name in arguments:
+                        arguments[arg.name] = []
+                    if value:
+                        arguments[arg.name].append(value)
```

Comment:
It changes the output of empty-value arguments like `/doc` from `{'doc': ['']}` to `{'doc': []}`